app/worker/smtp.go

```diff
@@ -55,6 +57,16 @@ func processSMTPTask(smtpTask *model.SMTPTask, job *model.Job,
 	from := processJobString(smtpTask.From, workflow, job)
 	subject := processJobString(smtpTask.Subject, workflow, job)
 	body := processJobString(smtpTask.Body, workflow, job)
```

Comment:
processJobString is the one that replaces the variables in the string with values (e.g., the workflow input parameters).
It should be called AFTER you read the body from the file. May I ask you why do we need to load the file from an external file? I've thought we are going to put the body of the email inside the workflow definition as a string, to keep it simple.

Another thing: please add a unit test, we are pretty good with code coverage already :)

Reply:
> processJobString is the one that replaces the variables in the string with values (e.g., the workflow input parameters).
> It should be called AFTER you read the body from the file.

It is fine with the way I use it in the workflows (the PR is going to be in workflows-enterprise, so thats why you did not see it, I am sorry I was waiting for the merge of this and previous commits).

line **59**: _smtpTask.Body is_ `'${workflow.input.body}'`
line **60**: _after processJobString body is_ `'@/etc/workflows/data/welcome_email_template.txt'` (from the workflow definition)
and then if it needs to be loaded from file, it is loaded, and on
line **68**: with _processJobString_ it is again filled, since in the mail body template we need to fill in the email address

it will all be clear once I file PRs with the workflows definitions.

> May I ask you why do we need to load the file from an external file? I've thought we are going to put the body of the email inside the workflow definition as a string, to keep it simple.

the above we discussed on slack, we decided to keep mail body loading from file.
> 
> Another thing: please add a unit test, we are pretty good with code coverage already :)

[sure!](https://github.com/mendersoftware/workflows/pull/24/commits/5ab8b3f4e79c9415c4092debed534f2fcaec0afb) I love a good coverage.